src/sentry/hybridcloud/models/outbox.py

```
@@ -219,11 +220,148 @@ def process_shard(self, latest_shard_row: OutboxBase | None) -> Generator[Outbox
+                yield None
+                return
+            else:
+                raise
+
+        tags["category"] = OutboxCategory(self.category).name
+        assert first_coalesced, "first_coalesced incorrectly set for non-empty coalesce group"
+        metrics.timing(
+            "outbox.coalesced_net_queue_time",
+            now.timestamp() - first_coalesced.date_added.timestamp(),
+            tags=tags,
+        )
+
+        # Yield the latest message for processing outside of a transaction
+        yield coalesced
+
+        # If the block didn't raise, delivery was successful, and we
+        # can delete messages to acknowledge them.
+        try:
+            deleted_count = 0
+            with transaction.atomic(using=using):
+                # The last message in a coalesce group should be preserved if we're
+                # skipping the shard, as we need to delay signals, and keep
+                # backlogs shorter, but not lose updates.
+                cleanup_ids = reserved_ids[:-1]
+                if cleanup_ids:
+                    self.objects.filter(id__in=cleanup_ids).delete()
+                    deleted_count = len(cleanup_ids)
+                if not self.should_skip_shard():
+                    deleted_count += 1
+                    coalesced.delete()
+
+            # refresh now, as signal handlers can take full seconds to run
+            now = timezone.now()
+            metrics.incr("outbox.processed", deleted_count, tags=tags)
+            metrics.timing(
+                "outbox.processing_lag",
+                now.timestamp() - first_coalesced.scheduled_from.timestamp(),
+                tags=tags,
+            )
+            metrics.timing(
+                "outbox.coalesced_net_processing_time",
+                now.timestamp() - first_coalesced.date_added.timestamp(),
+                tags=tags,
+            )
+        except Exception as err:
+            sentry_sdk.capture_exception(err)
```

The error isn't re-raised here and is swallowed. Is that intentional?

---

I'll remove the try/except we can let failures bubble up, the current code has no error traps around cleanup logic either.